### ISSUE TITLE
NodeRed Crashes when gateway is unrachble or full of connections

### ISSUE THREAD

**Issue by Maarten200** [Supergiovane/node-red-contrib-knx-ultimate#4]
My node red install on my Raspberry Pi completely crashes and becomes unreachable when my Gateway of my knxUltimate doesn't respond, wrong IP or connections full.
My gateway is an Hager TYF120.

If i disable the flow which contains knxUltimate node and restart node red it works perfectly again and if i enable the flow it crashes again after few moments.

if you want i can give an journalctl log, but gonna recreate the problem.

**Comment by Supergiovane** [comment]
Hello,
Will try to reproduce that... please hold on.

**Comment by Supergiovane** [comment]
Hello Maarten,
a little follow up.
I tried disconnecting my ETH/KNX Router, change multicast IP and so on, without a single issue. No crash and as soon as i've reconnected the ETH/KNX Router, the communication has automatically established and the datagram have started to flow regularly.
I've downloaded the user manual of the Hager TYF120 and discovered, that it's an IP interface without multicast capabilities.
The underlying knx.js API can automatically handle tunnelling instead of multicast, but there are no specif test about that, nor documentation.
Tunnelling involves TCP stack instead of UDP, with active connection handling and all related TCP stuffs.
Tomorrow i'll go to buy an EHT/KNX Interface to be able to test tunnelling connection and, if possibile, to fix your issue.
I suggest you to buy a ETH/KNX Router (when you change your Hager)  instead of an Interface, because you're not limited to a fixed numner of connection and you can use a multicast address via UDP, much more simply to setup.

**Comment by Supergiovane** [comment]
Hello Maarten,
second follow up.
Got an IP Interface (Eneretex IP Interface).
Tried to simulate issues on LAN - > no problem.
Tried to simulate issues on BUS - > the node has crashed, bringing node-red down.
I tried to simply disconnect the KNX TP bus connector and the problem appeared.

Thanks for the info. A pached for that is under development and will out in the version 1.1.8 as soon as possible.
